cp_test_nn/predict.py

```diff
@@ -1,23 +1,39 @@
+import sys
+
 from cp_test_nn import process
 from cp_test_nn.dataset import create_dataset
 
-def test_cv_success_rate(nn, tokens, cv_file):
+def predict(nn, tokens, cv_file):
     data = create_dataset(process.failures(cv_file), tokens)
     X = data['dataset']
     y = data['results']
 
+    # Statistics that get triggered when we have result data
     successes = 0
     false_negatives = 0
     false_positives = 0
-    total_count = len(X)
+    total_count = 0
+
     for i, item in enumerate(X):
         pred = nn.predict([item])
-        if pred == [y[i]]:
-            successes += 1
-        elif pred == [1]:
-            false_positives += 1
-        else:
-            false_negatives += 1
-    print('Success rate: ', successes/total_count)
-    print('False positives: ', false_positives/total_count)
-    print('False negatives: ', false_negatives/total_count)
+        sys.stdout.write("{0} {1} {2} {3} {4}\n".format(
+            pred and "FLAKE  " or "UNKNOWN",
```

Comment:
No problem :)

The neural network actually always gives a probability as a prediction, the scikit-learn predict method just rounds it to 0 or 1. So it definitely makes sense to look at probabilities. So we can definitely do something like "prob < 0.2 => not flake; prob > 0.8 => flake; 0.2 <= prob <= 0.8 => unknown". I haven't yet looked at the actual probabilities, so I'm not sure what would be reasonable margins. I can definitely look into this.